src/preferences/broadcastsettings.cpp

```diff
@@ -1,417 +1,135 @@
+        // Upgrade from mixxx.cfg format to XML (if required)
+        loadLegacySettings(defaultProfile);
+
+        saveProfile(defaultProfile);
+
+        m_profiles.push_back(defaultProfile);
+        setCurrentProfile(defaultProfile);
+    }
 }
 
-void BroadcastSettings::setStreamDesc(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamDesc), value);
-}
+void BroadcastSettings::saveProfile(BroadcastProfile* profile) {
+    if(!profile)
+        return;
 
-QString BroadcastSettings::getDefaultStreamDesc() const {
-    return kDefaultStreamDesc;
+    profile->save(filenameForProfile(profile));
 }
 
-QString BroadcastSettings::getStreamGenre() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamGenre),
-            getDefaultStreamGenre());
+QString BroadcastSettings::filenameForProfile(const QString& profileName) {
+    QString filename = profileName + QString(".bcp.xml");
+    return QDir(getProfilesFolder()).absoluteFilePath(filename);
 }
 
-void BroadcastSettings::setStreamGenre(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamGenre), value);
-}
+QString BroadcastSettings::filenameForProfile(BroadcastProfile* profile) {
+    if(!profile)
+        return QString();
 
-QString BroadcastSettings::getDefaultStreamGenre() const {
-    return kDefaultStreamGenre;
+    return filenameForProfile(profile->getProfileName());
 }
 
-QString BroadcastSettings::getStreamName() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamName),
-            getDefaultStreamName());
+QString BroadcastSettings::getProfilesFolder() {
+    QString profilesPath(m_pConfig->getSettingsPath());
+    profilesPath.append(QDir::separator() + QString(kProfilesSubfolder));
+    return profilesPath;
 }
 
-void BroadcastSettings::setStreamName(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamName), value);
-}
+void BroadcastSettings::setCurrentProfile(BroadcastProfile *profile) {
+    if(!profile)
+        return;
 
-QString BroadcastSettings::getDefaultStreamName() const {
-    return QString();
+    QString profileName = profile->getProfileName();
+    m_pConfig->setValue(ConfigKey(kConfigKey, kCurrentProfile), profileName);
 }
 
-bool BroadcastSettings::getStreamPublic() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamPublic), getDefaultStreamPublic());
+BroadcastProfile* BroadcastSettings::getCurrentProfile() {
+    QString currentProfile = m_pConfig->getValue(
+                                 ConfigKey(kConfigKey, kCurrentProfile),
+                                 kDefaultProfile);
+    return getProfileByName(currentProfile);
 }
 
-void BroadcastSettings::setStreamPublic(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamPublic), value);
+BroadcastProfile* BroadcastSettings::getProfileByName(const QString& profileName) {
+    for(BroadcastProfile* profile : m_profiles) {
```

Comment:
I just noticed that the profile name attribute in `BroadcastProfile` is useless in it (not used in load/save code and the XML document itself), so maybe name "keeping" should be the responsibility of `BroadcastSettings`?
That's why I initially used a list instead of a hashmap, to avoid having to synchronize names between the hashmap and the instances.